perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/gatherbev_plugin.hpp

```diff
@@ -0,0 +1,120 @@
+#define AUTOWARE__TENSORRT_BEVDET__GATHERBEV_PLUGIN_HPP_
+
+#include <NvInfer.h>
+#include <unistd.h>
+
+#include <cmath>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace
+{
+static const char * GATHERBEV_PLUGIN_NAME{"GatherBEV"};
+static const char * GATHERBEV_PLUGIN_VERSION{"1"};
+}  // namespace
+
+namespace nvinfer1
+{
+class GatherBEVPlugin : public IPluginV2DynamicExt
+{
+private:
+  const std::string name_;
+  std::string namespace_;
+  struct
+  {
+  } m_;
+
+public:
+  GatherBEVPlugin() = delete;
+  explicit GatherBEVPlugin(const std::string & name);
+  GatherBEVPlugin(const std::string & name, const void * buffer, size_t length);
+  ~GatherBEVPlugin();
+
+  // Method inherited from IPluginV2
+  const char * getPluginType() const noexcept override;
+  const char * getPluginVersion() const noexcept override;
+  int32_t getNbOutputs() const noexcept override;
+  int32_t initialize() noexcept override;
+  void terminate() noexcept override;
+  size_t getSerializationSize() const noexcept override;
+  void serialize(void * buffer) const noexcept override;
+  void destroy() noexcept override;
+  void setPluginNamespace(const char * pluginNamespace) noexcept override;
+  const char * getPluginNamespace() const noexcept override;
+
+  // Method inherited from IPluginV2Ext
+  DataType getOutputDataType(
+    int32_t index, DataType const * inputTypes, int32_t nbInputs) const noexcept override;
+  void attachToContext(
+    cudnnContext * contextCudnn, cublasContext * contextCublas,
+    IGpuAllocator * gpuAllocator) noexcept override;
+  void detachFromContext() noexcept override;
+
+  // Method inherited from IPluginV2DynamicExt
+  IPluginV2DynamicExt * clone() const noexcept override;
+  DimsExprs getOutputDimensions(
+    int32_t outputIndex, const DimsExprs * inputs, int32_t nbInputs,
+    IExprBuilder & exprBuilder) noexcept override;
+  bool supportsFormatCombination(
+    int32_t pos, const PluginTensorDesc * inOut, int32_t nbInputs,
+    int32_t nbOutputs) noexcept override;
+  void configurePlugin(
+    const DynamicPluginTensorDesc * in, int32_t nbInputs, const DynamicPluginTensorDesc * out,
+    int32_t nbOutputs) noexcept override;
+  size_t getWorkspaceSize(
+    const PluginTensorDesc * inputs, int32_t nbInputs, const PluginTensorDesc * outputs,
+    int32_t nbOutputs) const noexcept override;
+  int32_t enqueue(
+    const PluginTensorDesc * inputDesc, const PluginTensorDesc * outputDesc,
+    const void * const * inputs, void * const * outputs, void * workspace,
+    cudaStream_t stream) noexcept override;
+
+protected:
+  // To prevent compiler warnings
+  using nvinfer1::IPluginV2::enqueue;
+  using nvinfer1::IPluginV2::getOutputDimensions;
+  using nvinfer1::IPluginV2::getWorkspaceSize;
+  using nvinfer1::IPluginV2Ext::configurePlugin;
+};
+
+class GatherBEVPluginCreator : public IPluginCreator
+{
+private:
+  static PluginFieldCollection fc_;
```

Comment:
The use of static PluginFieldCollection and vector<PluginField> within the `GatherBEVPluginCreator` class should be reviewed. While there is no immediate issue seen in this snippet,  it's recommended to favor inline functions or other language constructs over preprocessor macros whenever possible to improve readability, maintainability, and reduce the risk of unexpected behavior. The guideline strongly discourages the use of macros, especially in headers.